src/test/java/com/thirdcc/webapp/web/rest/DebtResourceIT.java

```diff
@@ -125,61 +152,178 @@ public static Event createEventEntity() {
     }
 
     @AfterEach
     public void cleanUp() {
         debtRepository.deleteAll();
+        eventRepository.deleteAll();
+        eventAttendeeRepository.deleteAll();
+        eventCrewRepository.deleteAll();
     }
     
-    private void initDebtDB() {
-        debtRepository.saveAndFlush(debt);
+    private Debt initDebtDB() {
+        return debtRepository.saveAndFlush(debt);
     }
     
-    private void initEventDB() {
-        event = createEventEntity();
-        eventRepository.saveAndFlush(event);
+    private Event initEventDB() {
+        return eventRepository.saveAndFlush(event);
     }
-
+    
+    private EventAttendee initEventAttendeeDB() {
+        return eventAttendeeRepository.saveAndFlush(eventAttendee);
+    }
+    
+    private EventCrew initEventCrewDB() {
+        return eventCrewRepository.saveAndFlush(eventCrew);
+    }
+    
     @Test
-    public void getAllDebtsByEventId() throws Exception {
-        //TODO implement checking for admin and event crew
+    public void getAllDebtsByEventId_UserWithRoleAdmin() throws Exception {
         // Initialize the database
+        Event savedEvent = initEventDB();
+        eventAttendee.setEventId(savedEvent.getId());
+        EventAttendee savedEventAttendee = initEventAttendeeDB();
+        debt.setEventAttendeeId(savedEventAttendee.getId());
+        Debt savedDebt = initDebtDB();
+
+        // Get all the debt in the event
+        restDebtMockMvc.perform(get("/api/debts/event/{eventId}?sort=id,desc", savedEvent.getId()))
+            .andExpect(status().isOk())
+            .andExpect(content().contentType(MediaType.APPLICATION_JSON_UTF8_VALUE))
+            .andExpect(jsonPath("$.[*].id").value(hasItem(savedDebt.getId().intValue())))
+            .andExpect(jsonPath("$.[*].receiptId").value(hasItem(DEFAULT_RECEIPT_ID.intValue())))
+            .andExpect(jsonPath("$.[*].eventAttendeeId").value(hasItem(savedEventAttendee.getId().intValue())))
+            .andExpect(jsonPath("$.[*].amount").value(hasItem(DEFAULT_AMOUNT.doubleValue())))
+            .andExpect(jsonPath("$.[*].status").value(hasItem(DEFAULT_STATUS.toString())));
+    }
+    
+    @Test
+    public void getAllDebtsByEventId_UserWithRoleAdmin_NoDebtWithOpenStatus() throws Exception {
+        // Initialize the database
+        Event savedEvent = initEventDB();
+        eventAttendee.setEventId(savedEvent.getId());
+        EventAttendee savedEventAttendee = initEventAttendeeDB();
+        debt.setEventAttendeeId(savedEventAttendee.getId());
+        debt.setStatus(UPDATED_STATUS);
         initDebtDB();
-        initEventDB();
 
-        // Get all the debtList
+        // Get all the debt in the event
+        restDebtMockMvc.perform(get("/api/debts/event/{eventId}?sort=id,desc", savedEvent.getId()))
+            .andExpect(status().isOk())
+            .andExpect(content().contentType(MediaType.APPLICATION_JSON_UTF8_VALUE))
+            .andExpect(content().string("[]"));
```

Comment:
why not check if the response is empty size
like `.andExpect(jsonPath("$", hasSize(0)));`

Reply:
Resolved by https://github.com/Five-Fishes/Club-Management-Spring-Server/pull/59/commits/e02ef4a5ce9bccd4ecd8c5d82e54b4f05c6073f4